File: lib/models/sub_court.dart
class SubCourt {
  final int? id;
  final String name;
  final int courtId;

  SubCourt({
    this.id,
    required this.name,
    required this.courtId,
  });

  factory SubCourt.fromJson(Map<String, dynamic> json) {
    return SubCourt(
      id: json['id'],
      name: json['name'],
      courtId: json['courtId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'courtId': courtId,
    };
  }
}

File: backend/BadmintonBookingAPI/Models/SubCourt.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace BadmintonBookingAPI.Models;

public class SubCourt
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [Required]
    public int CourtId { get; set; }

    [ForeignKey("CourtId")]
    [JsonIgnore]
    public Court? Court { get; set; }

    [JsonIgnore]
    public ICollection<TimeSlot> TimeSlots { get; set; } = new List<TimeSlot>();
}
